ChatQnA/benchmark/performance/kubernetes/intel/gaudi/README.md

```diff
@@ -0,0 +1,211 @@
+```
+#### Case 2: Baseline Deployment without Rerank
+
+```bash
+python deploy.py --hftoken $HFTOKEN --modeldir $MODELDIR --num-nodes 2
+```
+#### Case 3: Tuned Deployment with Rerank
+
+```bash
+python deploy.py --hftoken $HFTOKEN --modeldir $MODELDIR --num-nodes 2 --with-rerank --tuned
+```
+
+## Benchmark
+
+### Test Configurations
+
+| Key      | Value   |
+| -------- | ------- |
+| Workload | ChatQnA |
+| Tag      | V1.0    |
+
+Models configuration
+| Key | Value |
+| ---------- | ------------------ |
+| Embedding | BAAI/bge-base-en-v1.5 |
+| Reranking | BAAI/bge-reranker-base |
+| Inference | Intel/neural-chat-7b-v3-3 |
+
+Benchmark parameters
+| Key | Value |
+| ---------- | ------------------ |
+| LLM input tokens | 1024 |
+| LLM output tokens | 128 |
+
+Number of test requests for different scheduled node number:
+| Node count | Concurrency | Query number |
+| ----- | -------- | -------- |
+| 1 | 128 | 640 |
+| 2 | 256 | 1280 |
+| 4 | 512 | 2560 |
+
+More detailed configuration can be found in configuration file [benchmark.yaml](./benchmark.yaml).
+
+### Test Steps
+
+#### Upload Retrieval File
+
+Before running tests, upload a specified file to make sure the llm input have the token length of 1k.
+
+Run the following command to check the cluster ip of dataprep.
+
+```bash
+kubectl get svc
+```
+
+Substitute the `${cluster_ip}` into the real cluster ip of dataprep microservice as below.
+
+```log
+dataprep-svc   ClusterIP   xx.xx.xx.xx    <none>   6007/TCP   5m   app=dataprep-deploy
+```
+
+Run the cURL command to upload file:
+
+```bash
+cd GenAIEval/evals/benchmark/data
+# RAG with Rerank
+curl -X POST "http://${cluster_ip}:6007/v1/dataprep" \
+     -H "Content-Type: multipart/form-data" \
+     -F "files=@./upload_file.txt" \
+     -F "chunk_size=3800"
+# RAG without Rerank
+curl -X POST "http://${cluster_ip}:6007/v1/dataprep" \
+     -H "Content-Type: multipart/form-data" \
+     -F "files=@./upload_file_no_rerank.txt"
+```
+
+#### Run Benchmark Test
+
+Before the benchmark, we can configure the number of test queries and test output directory by:
+
+```bash
+export USER_QUERIES="[640, 640, 640, 640]"
+export TEST_OUTPUT_DIR="/home/sdp/benchmark_output/node_1"
+```
+
+And then run the benchmark by:
+
+```bash
+bash benchmark.sh -n 1
+```
+
+The argument `-n` refers to the number of test nodes. Note that necessary dependencies will be automatically installed when running benchmark for the first time.
+
+##### Data collection
+
+All the test results will come to this folder `/home/sdp/benchmark_output/node_1` configured by the environment variable `TEST_OUTPUT_DIR` in previous steps.
+
+#### Benchmark multiple nodes
+
+##### 2 node
+
+```bash
+export USER_QUERIES="[1280, 1280, 1280, 1280]"
+export TEST_OUTPUT_DIR="/home/sdp/benchmark_output/node_2"
+```
+
+And then run the benchmark by:
+
+```bash
+bash benchmark.sh -n 2
+```
+
+##### 4 node
+
+```bash
+export USER_QUERIES="[2560, 2560, 2560, 2560]"
```

Comment:
To make the readme simple, we don't need this part. This will auto set by benchmark.sh base on -n.